52_wa_diff_shipment.py

```diff
@@ -0,0 +1,263 @@
+        "Product_Name",
+        "Ingredient_Name",
+        "Measure",
+        "MME_Conversion_Factor",
+        "dos_str",
+        "Year",
+        "Month",
+        "StateFIPS",
+        "StateName",
+        "CountyFIPS",
+        "state_abbrev",
+        "FIP_unique",
+        "Population",
+        "county_test",
+    ]
+]
+
+# creating a copy of reduced dataset of prescriptions and converting some of the attributes to appropriate data type
+
+prescriptions_reduced_copy = prescriptions_reduced.copy()
+
+prescriptions_reduced_copy["Year"] = prescriptions_reduced_copy["Year"].astype("int64")
+prescriptions_reduced_copy["DRUG_CODE"] = prescriptions_reduced_copy[
+    "DRUG_CODE"
+].astype("int64")
+prescriptions_reduced_copy["Month"] = prescriptions_reduced_copy["Month"].astype(
+    "int64"
+)
+prescriptions_reduced_copy["Population"] = prescriptions_reduced_copy[
+    "Population"
+].astype("int64")
+
+# creating a dataset that has all the drug prescriptions in the state of Florida
+
+washington_prescriptions = prescriptions_reduced_copy[
+    prescriptions_reduced_copy["StateName"] == "Washington"
+]
+# washington_prescriptions_copy = washington_prescriptions.copy()
+washington_prescriptions["shipment_per_100k"] = (
+    (
+        washington_prescriptions["dos_str"]
+        * washington_prescriptions["DOSAGE_UNIT"]
+        * washington_prescriptions["MME_Conversion_Factor"]
+    )
+    / (washington_prescriptions["Population"])
+    * 100000
+)
+
+washington_prescriptions
+
+# calculating and displaying total number of drug prescriptions that took place in washington. results are grouped and displayed year and county wise
+
+washington_prescriptions_result = (
+    washington_prescriptions.groupby(["Year", "county_test"])["shipment_per_100k"]
+    .sum()
+    .reset_index()
+)
+# washington_prescriptions_result = washington_prescriptions.groupby(["Year"])["shipment_per_100k"].sum().reset_index()
+
+washington_prescriptions_result.head()
+
+## Diff-in-Diff
+
+other_states_shipment = prescriptions_reduced_copy[
+    (prescriptions_reduced_copy["StateName"] != "Washington")
+]
+# other_states_shipment = prescriptions_reduced_copy[(prescriptions_reduced_copy["StateName"] == "Alabama")|(prescriptions_reduced_copy["StateName"] == "Georgia")|(prescriptions_reduced_copy["StateName"] == "Mississippi")|(prescriptions_reduced_copy["StateName"] == "South Carolina")|(prescriptions_reduced_copy["StateName"] == "Tennessee")]
+
+other_states_shipment
+
+other_states_shipment_copy = other_states_shipment.copy()
+
+# other_states_shipment_copy["shipment_per_100k"] = (other_states_shipment_copy["QUANTITY"] / other_states_shipment_copy["Population"]) * 100_000
+other_states_shipment_copy["shipment_per_100k"] = (
+    (
+        other_states_shipment_copy["dos_str"]
+        * other_states_shipment_copy["DOSAGE_UNIT"]
+        * other_states_shipment_copy["MME_Conversion_Factor"]
+    )
+    / (other_states_shipment_copy["Population"])
+    * 100000
+)
```

Comment:
You will need to revise the formula. The reason is that using this formula for calculating the volume of opioid shipped, we have been getting around 80K Null Values, which was affecting the analysis.

The new formula that you would want to use -

other_states_shipment_copy["shipment_per_100k"] = (
(
other_states_shipment_copy["CALC_BASE_WT_IN_GM"]*
other_states_shipment_copy["MME_Conversion_Factor"]
)
/ (other_states_shipment_copy["Population"])
*100000
)

This greatly reduces the number of null values present for total volume of opioid shipped.

Reply:
Again, so much appreciation on this. Already revised in my new ipynb notebook in the main branch

Reply:
Good job on devising an easier way to calculate total opioid shipped!